pandas/io/excel/_base.py

```diff
@@ -837,14 +850,16 @@ class ExcelFile:
     from pandas.io.excel._pyxlsb import _PyxlsbReader
     from pandas.io.excel._xlrd import _XlrdReader
 
-    _engines = {
+    _engines: Mapping[str, Any] = {
         "xlrd": _XlrdReader,
         "openpyxl": _OpenpyxlReader,
         "odf": _ODFReader,
         "pyxlsb": _PyxlsbReader,
     }
 
-    def __init__(self, path_or_buffer, engine=None):
+    def __init__(
+        self, path_or_buffer, storage_options: StorageOptions = None, engine=None
```

Comment:
Put storage_options after engine, for users passing that as a position?

Reply:
done